### ISSUE TITLE
Empty logs after a failed import statement

### ISSUE THREAD

**Issue by caleb-johnson** [Qiskit/qiskit-serverless#1105]
Job id 'd5234ef5-2873-44b9-a205-4fad9aab1fdc' had an import statement that wasn't available in the deployed Qiskit version (qiskit.circuit.lilbrary.UnitaryOverlap)

The problem is that for this particular job, job.logs() returned empty, instead of returning the import error. The same thing happened for several runs before this.

Job ID '0b3eeb8f-55c0-42ff-87c3-b3c285683380' properly displayed the error in the logs, but one thing I did different in this job was that I embedded print statements throughout the script. I'm not sure if that played a part, or if it randomly worked as expected in this job.

**Comment by caleb-johnson** [comment]
`'fd64febc-0c45-48cf-837b-6c121deea404'

result and logs were empty for this script when the same script successfully ran on previous iteration

**Comment by caleb-johnson** [comment]
> @caleb-johnson can you share a version of this script so we can try to reproduce?

I don't believe the contents of the script, or any specific issue with the source code, was causing this. It seemed like intermittently the job logs would be `None`, regardless of what was contained in the logs.

I had a hypothesis that maybe I was spamming the jupyter notebook cells too fast and ~4 API calls were being submitted simultaneously (submit, result, job_id, logs) and it was causing the logs to behave strangely. I'm not sure if that is a possibility, but a few times it errored in this way I did notice I had run multiple cells/API calls within less than a second

**Comment by caleb-johnson** [comment]
```
provider = Provider(token="", host="https://middleware.quantum-computing.test.cloud.ibm.com")
serverless = QuantumServerless(provider)

program = Program(title="program", entrypoint="program.py", working_dir="./src/")

job = serverless.run(program)

job.job_id

job.status()

job.logs()
```

This is the notebook script which intermittently/incorrectly returned empty logs. The contents of the source code didn't seem to play a role. Each spaced represents a new cell in the notebook (in case this is some kind of jupyter bug).

**Comment by psschwei** [comment]
Thanks @caleb-johnson , will try to break stuff shortly :smile:

**Comment by Tansito** [comment]
We didn't receive any report related with this so let's assume this was fixed in once of our several developments 👍
